weather.py

```diff
@@ -0,0 +1,414 @@
+# Imports written in alphabetical order
+from os import path
+from csv_reader import CSVReader
+import sys
+from termcolor import colored
+import zipfile
+
+
+class Weatherman:  # Weatherman Application in one class
+
+    file_name_prefix = "Murree_weather_"  # The part of the file name that is same for all files
+
+    # List with months to access by numerical form
+    months = ["Jan", "Feb", "Mar",
+              "Apr", "May", "Jun",
+              "Jul", "Aug", "Sep",
+              "Oct", "Nov", "Dec"
+              ]
+
+    # Class Constructor
+    def __init__(self, path, highest_argument, average_argument, chart_form_argument, chart_form_combined):
+        self.__path = path
+        self.__highest_arguments = highest_argument
+        self.__average_arguments = average_argument
+        self.__chart_form_arguments = chart_form_argument
+        self.__chart_form_combined = chart_form_combined
+
+        # path common in all files
+        self.files_path_prefix = self.__path + "/weatherfiles/" + Weatherman.file_name_prefix
+
+        # Report Number to print
+        self.report_num = 1
+
+    # The method main calls to start the code
+    def run(self):
+        self.extract_files()  # call function to extract weatherman zip to path
+
+        # All arguments will be checked to ensure multiple arguments accepted
+        if self.__highest_arguments is not None:  # check if -e argument was set
+            for highest_argument in self.__highest_arguments:  # catering multiple inputs argument for multiple reports
+                self.highest_argument_handler(highest_argument)  # Calculating results one input at a time
+
+        if self.__average_arguments is not None:  # check if -a argument was set
+            for average_argument in self.__average_arguments:
+                self.average_argument_handler(average_argument)
+
+        if self.__chart_form_arguments is not None:  # check if -c argument was set
+            for chart_form_argument in self.__chart_form_arguments:
+                self.chart_argument_handler(chart_form_argument)
+
+    # Method to extract weather files to specified path
+    def extract_files(self):
+        # Extract the weatherman.zip file to path
+        with zipfile.ZipFile("weatherfiles.zip", "r") as zip_ref:
+            zip_ref.extractall(self.__path)
+
+    # Method that calculates the read data
+    def highest_argument_handler(self, current_argument):
+
+        is_valid = self.validate_year(current_argument)
+
+        if is_valid:
+            is_file = self.files_exists_in_year(current_argument)
+
+            if is_file:
+                result = self.calculate_highest(current_argument)
+
+                # Function that prints report according to the mode provided
+                self.generate_report_highest(result)
+            else:
+                print(f"Files for the year {current_argument} are not in the system.")
```

Comment:
We can use guard statements to simplify this flow. Instead of writing nested conditions, we can write a reverse `if` statement to stop the flow if the given information is not valid and we can't proceed further. It simplifies the logic and improves readability. Please follow the same pattern at other places too.
```suggestion
        is_valid = self.validate_year(current_argument)
        if not is_valid:
            print("Please enter a valid year to calculate the yearly highest report")
            return
            
        has_files = self.files_exists_in_year(current_argument)
        if not has_files:
            print(f"Files for the year {current_argument} are not in the system.")
            return
            
        result = self.calculate_highest(current_argument)
        
        # Function that prints report according to the mode provided
        self.generate_report_highest(result)
```

Reply:
done